---
title: "Lab 12 Genome Informatics HW Pop Analysis"
author: "Daniel Gurholt (PID: A16767491)"
format: pdf
---

## Section 4: Population Scale Analysis [HOMEWORK]

One sample is obviously not enough to know what is happening in a population. You are
interested in assessing genetic differences on a population scale. So, you processed about
~230 samples and did the normalization on a genome level. Now, you want to find whether
there is any association of the 4 asthma-associated SNPs (rs8067378â€¦) on ORMDL3
expression.

This is the final file you got ( https://bioboot.github.io/bggn213_W19/classmaterial/
rs8067378_ENSG00000172057.6.txt ). The first column is sample name, the second
column is genotype and the third column are the expression values.

>Q13: Read this file into R and determine the sample size for each genotype and their
corresponding median expression levels for each of these genotypes.

#Reading in genotype data

```{r}
genedat<- read.table("gene data.txt")
head(genedat)
```

#Determining sample size for the samples

```{r}
nrow(genedat)
```

There is a total of 462 gene samples in this given data set.

```{r}
table(genedat$geno)
```

There are 108 gene samples that are homozygous A|A, 233 gene samples that are heterozygous A|G, and 121 samples homozygous for G|G.

#Determining medians for each of the 3 genotypes present in the dataset

```{r}
library(dplyr)
AA<- genedat %>%
  filter(geno == "A/A")
head(AA)

summary(AA)
```

The median expression levels for the A/A genotype is 31.25

```{r}
library(dplyr)
AG<- genedat %>%
  filter(geno == "A/G")
head(AG)

summary(AG)
```

The median expression levels for the A/G genotype is 25.065

```{r}
library(dplyr)
GG<- genedat %>%
  filter(geno == "G/G")
head(GG)

summary(GG)
```

The median expression levels for the G/G genotype is 20.074


>Q14: Generate a boxplot with a box per genotype, what could you infer from the relative
expression value between A/A and G/G displayed in this plot? Does the SNP effect the
expression of ORMDL3?

```{r}
library(ggplot2)

Box<- ggplot(genedat) +
  aes(x=geno, y=exp, fill=geno) +
  geom_boxplot(notch=T)
Box
```

Looking at the resulting boxplot shows us that having the G/G genotype leads to an overall less expression of ORMDL3 than if the A/A genotype is present since expression levels are much higher in A/A compared to G/G. From this boxplot, I could conclude that the SNP does effect the expression of ORMDL3 depending on if a certain genotype is expressed or not.





